src/Provider/AsymmetricCookieSettingsProvider.php

```diff
@@ -0,0 +1,230 @@
+<?php
```

Comment:
Class is very similar to `CookieSettingsProvider`. Maybe we could move some `CookieSettingsProvider` code to protected methods? so that `AsymmetricCookieSettingsProvider` could use the same old cookie provider.